consensus/broadcast/shardChainMessenger.go

```diff
@@ -142,34 +142,56 @@ func (scm *shardChainMessenger) BroadcastBlockDataLeader(
 	transactions map[string][][]byte,
 	pkBytes []byte,
 ) error {
-	if check.IfNil(header) {
-		return spos.ErrNilHeader
-	}
-	if len(miniBlocks) == 0 {
+	if miniBlocks == nil {
 		return nil
 	}
-
-	headerHash, err := core.CalculateHash(scm.marshalizer, scm.hasher, header)
+	dtb, err := scm.prepareDataToBroadcast(header, miniBlocks, transactions, pkBytes)
+	if err != nil {
+		return err
+	}
+	err = scm.delayedBlockBroadcaster.SetLeaderData(dtb.delayedBroadcastData)
 	if err != nil {
 		return err
 	}
 
-	metaMiniBlocks, metaTransactions := scm.extractMetaMiniBlocksAndTransactions(miniBlocks, transactions)
+	go scm.BroadcastBlockData(dtb.metaMiniBlocks, dtb.metaTransactions, pkBytes, common.ExtraDelayForBroadcastBlockInfo)
```

Comment:
maybe add a TODO to analyze how we can improve this, similar to equivalent proofs broadcast

Reply:
right, added in our checklist, will add the todo here as well.